File: model/article.go
package model

import "gorm.io/gorm"

// Article 文章数据
type Article struct {
	gorm.Model
	UserID   uint   // 用户id
	UserName string // 用户名
	Title    string // 文章标题
	SubTitle string // 文章简介
	ArtType  string // 文章类型
	HeadImg  string // 首页图
	Content  string // 文章内容
	Likes    uint   // 点赞数量
	Share    uint   // 分享数量
	Super    bool   // 加精
}

// ArticleDto 文章数据传输
type ArticleDto struct {
	CreateAt string `json:"create"`   // 创建时间
	UpdateAt string `json:"update"`   // 修改事件
	ArtID    uint   `json:"artid"`    // 文章id
	UserID   uint   `json:"userid"`   // 用户id
	UserName string `json:"username"` // 用户名
	Title    string `json:"title"`    // 文章标题
	SubTitle string `json:"subtitle"` // 文章简介
	ArtType  string `json:"arttype"`  // 文章类型
	HeadImg  string `json:"headimg"`  // 首页图
	Content  string `json:"content"`  // 文章内容
	Likes    uint   `json:"likes"`    // 点赞数量
	Share    uint   `json:"share"`
	Super    bool   `json:"super"`
}

File: web/src/article/article.js
$(function () {
  // URL解析
  $.getUrlParam = function (name) {
    var reg = new RegExp('(^|&)' + name + '=([^&]*)(&|$)')
    var r = window.location.search.substr(1).match(reg)
    if (r != null) return unescape(r[2])
    return null
  }

  var artid = $.getUrlParam('artid')

  console.log(artid)
  if (artid == null) {
    window.location.href = '/'
    return
  }
  // 发起请求
  infoURL = 'http://127.0.0.1:8080/getart'
  $.ajax(infoURL, {
    method: 'POST',
    data: {
      artid: artid,
    },
    // 得到响应的处理
    success: (data) => {
      if (data.code != '200') {
        $('#userNULL').removeClass('d-none')
      }
      $('#headimg').attr('src', data.data.user.headimg)
      $('#username').text(data.data.user.name)
      $('#artid').text(data.data.article.artid)
      $('#arttype').text(data.data.article.arttype)
      $('#createTime').text(data.data.article.create)
      $('#updateTime').text(data.data.article.update)
      $('#likesg').text(data.data.article.likes)
      $('#share').text(data.data.article.share)
      $('#title').text(data.data.article.title)
      $('#artimg').attr('src', data.data.article.headimg)
      $('#content').html(data.data.article.content.replace('"', '"'))
    },
    error: () => {
      alert('服务器可能正在维护，请刷新重试或联系管理员')
    },
  })
})
